### PR TITLE
Updated waveica for new outputs from recetox-aplcms

### LINKED ISSUE #365: check and update waveica for new outputs from recetox-aplcms and RAMClustR

### PR DESCRIPTION
**Pull request by zargham-ahmad**
This PR includes the following:

- Added data transpose option so the new outputs from recetox-aplcms can be used
- Added verify_column_types function to check if the columns are of the correct type
- Added test and testdata for transpose case
- Fixed some param calls in the test (filter_length -> wavelet_length)
- added group var in waveica_singlebatch function which was not there but used
- Bump version

closes #365

### CODE REVIEW

**Review by hechth — COMMENTED** [review]
LGTM - only add WaveICA to bio.tools and then add the crossref here and we're done :)

**Comment by zargham-ahmad** [comment]
> LGTM - only add WaveICA to bio.tools and then add the crossref here and we're done :)

added in [a54098b](https://github.com/RECETOX/galaxytools/pull/378/commits/a54098b338e6df900ec886f7662ba47938a6d78c)

File: tools/waveica/waveica.xml [hunk 1]
```diff
@@ -1,8 +1,9 @@
-<tool id="waveica" name="WaveICA" version="@TOOL_VERSION@+galaxy2" python_template_version="3.5">
+<tool id="waveica" name="WaveICA" version="@TOOL_VERSION@+galaxy3" python_template_version="3.5">
     <description>removal of batch effects for untargeted metabolomics data</description>
     <macros>
         <import>macros.xml</import>
     </macros>
+    <expand macro='annotation'/>
```
Comment (xtrojak) [inline tools/waveica/waveica.xml:6]: Keep consistent quotation marks type.

```suggestion
    <expand macro="annotation"/>
```

File: tools/waveica/waveica.xml [hunk 2]
```diff
@@ -61,10 +64,12 @@
...
             </when>
             <when value="2">
+                <expand macro="input_data"/>
                 <section name="input_metadata" title="Input metadata table" expanded="true">
                     <param name="metadata" label="Input metadata" type="data" format="csv,tsv,parquet" help="" />
-                </section> 
-                <expand macro="input_data"/>
+                </section>
+                <param name = "transpose_feature_table" label="Transpose feature table" type="boolean" checked="false" 
+                truevalue="TRUE" falsevalue="FALSE" help="Transpose if feature names are -> column 1 and sampleNames are column" />
```
Comment (xtrojak) [inline tools/waveica/waveica.xml:72]: Somehow the `help` doesn't help.

**Review by xtrojak — APPROVED** [review]
Except minor issues looks good to me.

File: tools/waveica/waveica.xml (hunk 2 above)
Comment (hechth) [reply tools/waveica/waveica.xml:72]: True. This could be improved a bit :D @zargham-ahmad

Comment (zargham-ahmad) [reply tools/waveica/waveica.xml]: Updated in [288aaae](https://github.com/RECETOX/galaxytools/pull/378/commits/288aaae211afc339670bc205845a4393665a8455)

Comment (zargham-ahmad) [reply tools/waveica/waveica.xml]: Is it better now? or maybe it seems more complicated now :D

Comment (xtrojak) [reply tools/waveica/waveica.xml]: Yeah, now it's actually a sentence :D but still takes a minute to understand. I'm thinking of it's even needed to explain, I suppose "transpose" is pretty clear. Anyway, maybe a more straightforward formulation: "Swap sample names with feature names as column headers (to fit aplcms outputs)."

Comment (zargham-ahmad) [reply tools/waveica/waveica.xml]: Yes, this looks good as well

**Pull request merged**